app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -16,62 +16,74 @@
 
 package uk.gov.hmrc.cgtpropertydisposals.service
 
+import akka.pattern.ask
+import akka.util.Timeout
 import cats.data.EitherT
-import cats.instances.future._
 import com.google.inject.{ImplementedBy, Inject, Singleton}
-import play.api.http.Status._
-import uk.gov.hmrc.cgtpropertydisposals.connectors.TaxEnrolmentConnector
-import uk.gov.hmrc.cgtpropertydisposals.models.{Address, EnrolmentRequest, Error, KeyValuePair, SubscriptionDetails}
-import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService.TaxEnrolmentResponse
-import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService.TaxEnrolmentResponse.TaxEnrolmentCreated
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.AttemptTaxEnrolmentAllocationToGroup
+import uk.gov.hmrc.cgtpropertydisposals.models.{Error, TaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.modules.TaxEnrolmentRetryProvider
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService.TaxEnrolmentDatabaseInsertResponse
+import uk.gov.hmrc.cgtpropertydisposals.util.Logging
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-
+import scala.concurrent.duration._
 @ImplementedBy(classOf[TaxEnrolmentServiceImpl])
 trait TaxEnrolmentService {
-  def allocateEnrolmentToGroup(cgtReference: String, subscriptionDetails: SubscriptionDetails)(
+  def allocateEnrolmentToGroup(taxEnrolmentRequest: TaxEnrolmentRequest)(
     implicit hc: HeaderCarrier
-  ): EitherT[Future, Error, TaxEnrolmentResponse]
-}
-
-object TaxEnrolmentService {
-
-  sealed trait TaxEnrolmentResponse
-
-  object TaxEnrolmentResponse {
-    case object TaxEnrolmentCreated extends TaxEnrolmentResponse
-  }
-
+  ): EitherT[Future, Error, Unit]
 }
 
 @Singleton
-class TaxEnrolmentServiceImpl @Inject()(connector: TaxEnrolmentConnector) extends TaxEnrolmentService {
+class TaxEnrolmentServiceImpl @Inject()(
+  taxEnrolmentRetryProvider: TaxEnrolmentRetryProvider
+) extends TaxEnrolmentService
+    with Logging {
 
-  override def allocateEnrolmentToGroup(cgtReference: String, subscriptionDetails: SubscriptionDetails)(
+  override def allocateEnrolmentToGroup(taxEnrolmentRequest: TaxEnrolmentRequest)(
     implicit hc: HeaderCarrier
-  ): EitherT[Future, Error, TaxEnrolmentResponse] = {
-
-    val enrolmentRequest = subscriptionDetails.address match {
-      case Address.UkAddress(line1, line2, line3, line4, postcode) =>
-        EnrolmentRequest(List(KeyValuePair("Postcode", postcode)), List(KeyValuePair("CGTPDRef", cgtReference)))
-      case Address.NonUkAddress(line1, line2, line3, line4, maybePostcode, countryCode) =>
-        EnrolmentRequest(List(KeyValuePair("CountryCode", countryCode.code)), List(KeyValuePair("CGTPDRef", cgtReference)))
-    }
+  ): EitherT[Future, Error, Unit] = {
+    implicit val timeout: Timeout = Timeout(5.seconds)
```

Comment:
it might be a good idea to have the timeout configurable so that you can control it in the unit test when you test what happens if there's no reply from the ask without having to wait 5 seconds